pilot_experiment/nbconverted/2_explore_data.py

```diff
@@ -0,0 +1,332 @@
+# All gene correlations
+real_all_corr_score = real_all_corr.values[np.triu_indices(n=len(real_all_corr), k=1)]
+all_labels = np.repeat(['all'],len(real_all_corr_score))
+
+real_all_corr_df = pd.DataFrame(data={'corr_score': real_all_corr_score,
+                                     'group': all_labels})
+
+print(real_all_corr_df.shape)
+print('mean ', np.mean(real_all_corr_score))
+print('median ', np.median(real_all_corr_score))
+print('var ', np.var(real_all_corr_score))
+real_all_corr_df.head()
+
+
+# In[19]:
+
+
+# Flatten and get only upper triangle values from correlation matrix
+# Shuffled gene correlations
+shuffled_all_corr_score = shuffled_all_corr.values[np.triu_indices(n=len(shuffled_all_corr), k=1)]
+shuffled_all_labels = np.repeat(['shuffled'],len(real_all_corr_score))
+
+shuffled_all_corr_df = pd.DataFrame(data={'corr_score': shuffled_all_corr_score,
+                                          'group': shuffled_all_labels})
+
+print(shuffled_all_corr_df.shape)
+print('mean ', np.mean(shuffled_all_corr_score))
+print('median ', np.median(shuffled_all_corr_score))
+print('var ', np.var(shuffled_all_corr_score))
+shuffled_all_corr_df.head()
+
+
+# In[20]:
+
+
+# Create df
+corr_scores_df = pd.concat([real_core_corr_df,
+                            real_acc_corr_df,
+                            real_core_acc_corr_df,
+                            real_all_corr_df,
+                            shuffled_all_corr_df],
+                           ignore_index=True)
+
+print(corr_scores_df.shape)
+corr_scores_df.head()
+
+
+# In[21]:
+
+
+# Plot all correlation scores
+sns.boxplot(data=corr_scores_df,
+           x='group',
+           y='corr_score',
+           palette='Set3').set_title('Distribution of correlation scores per group')
+
+
+# In[48]:
+
+
+# Distribution plot for core genes
+sns.distplot(real_core_corr_score, label='core', color='red')
+sns.distplot(real_acc_corr_score, label='accessory', color='blue')
+sns.distplot(real_core_acc_corr_score, label='core-accessory', color='purple')
+sns.distplot(shuffled_all_corr_score, label='shuffled', color='grey')
+
+plt.legend(prop={'size': 12})
+plt.title('Density of correlation scores per group')
+plt.ylabel('Density')
+
+
+# **Some sanity checks**
+# * Shuffled dataset has very little correlation between genes, as expected since we have disrupted the inherent relationships between genes through our permutation process
+# * Since core genes comprise 97% of the total genes, the mean correlation for all genes is the same as the core gene set
+# 
+# **Other observations**
+# * The mean correlation of accessory genes only is 0.2295, which is higher compared to the mean correlation of core genes, 0.01037.
+# * The mean correlation of accessory-core genes is -0.036, which is lower compared to core-only and accessory only groups.
+# 
+# * Looking at the density plot for the accessory gene correlation scores, the scores are shifted to the right.
```

Comment:
I think being able to have a quantitative measure for the difference in trends will definitely be something we will want to do if the downstream results look promising! 

Just to clarify, what do you mean by `1000 random samples of genes from the whole gene set having the same size as the accessory gene set,`? I guess the accessory genes will always be the same, but do you mean select different samples from different experiments? So this will look to see if the trend we observe in this small subset is robust?

Reply:
> do you mean select different samples from different experiments?

Not exactly, I was just thinking you could select n random genes from the union of all the core and accessory genes (where n is the size of the accessory gene set), calculate the mean correlation of those genes, then do that 1000 times to get a "null distribution" of means.

The idea is that the accessory genes are a non-random sample of genes from all of the possible genes in PAO1/PA14, and you want to see if the mean correlation is higher than you would expect by chance if you took a random sample of genes. To do that, you need to approximate the null distribution of all possible random samples having the same size as the accessory genes, which you can do just by sampling uniformly a bunch of times.

> So this will look to see if the trend we observe in this small subset is robust?

Right, I think what I'm proposing would be one way of getting at this (although there are probably others).

Does that make any more sense? Happy to chat more about this if you're still confused.